File: src/features/typing/data/go/longest_repeating_character_replacement.go
package main

func characterReplacement(s string, k int) int {
	count := make(map[byte]int)
	maxCount, left, best := 0, 0, 0
	for right := 0; right < len(s); right++ {
		count[s[right]]++
		if count[s[right]] > maxCount {
			maxCount = count[s[right]]
		}
		for right-left+1 > maxCount+k {
			count[s[left]]--
			left++
		}
		if right-left+1 > best {
			best = right - left + 1
		}
	}
	return best
}

File: src/features/typing/data/ts/longest_repeating_character_replacement.ts
function characterReplacement(s: string, k: number): number {
    const count = new Map<string, number>();
    let maxCount = 0;
    let left = 0;
    let best = 0;
    for (let right = 0; right < s.length; right++) {
        const next = (count.get(s[right]) ?? 0) + 1;
        count.set(s[right], next);
        if (next > maxCount) {
            maxCount = next;
        }
        while (right - left + 1 > maxCount + k) {
            count.set(s[left], count.get(s[left])! - 1);
            left++;
        }
        if (right - left + 1 > best) {
            best = right - left + 1;
        }
    }
    return best;
}
